schema/gen/go/genMap.go

```diff
@@ -0,0 +1,631 @@
+	//   and may additionally incur a memcpy if the maybe for the value type doesn't use pointers internally).
+	// REVIEW: is there a way we can make this less twisty?  it is VERY unfortunate if the user has to know what sort of map it is to know which method to prefer.
+	//  Maybe the Lookup method on maps that have nullable values should just always have a MaybeT return type?
+	//   But then this means the Lookup method doesn't "need" an error as part of its return signiture, which just shuffles differences around.
+	doTemplate(`
+		func (n *_{{ .Type | TypeSymbol }}) LookupMaybe(k {{ .Type.KeyType | TypeSymbol }}) Maybe{{ .Type.ValueType | TypeSymbol }} {
+			v, ok := n.m[*k]
+			if !ok {
+				return &_{{ .Type | TypeSymbol }}__valueAbsent
+			}
+			{{- if .Type.ValueIsNullable }}
+			return v
+			{{- else}}
+			return &_{{ .Type.ValueType | TypeSymbol }}__Maybe{
+				m: schema.Maybe_Value,
+				v: {{ if not (MaybeUsesPtr .Type.ValueType) }}*{{end}}v,
+			}
+			{{- end}}
+		}
+
+		var _{{ .Type | TypeSymbol }}__valueAbsent = _{{ .Type.ValueType | TypeSymbol }}__Maybe{m:schema.Maybe_Absent}
+
+		// TODO generate also a plain Lookup method that doesn't box and alloc if this type contains non-nullable values!
+	`, w, g.AdjCfg, g)
+	// FUTURE: also a speciated iterator?
+}
+
+func (g mapGenerator) EmitNativeBuilder(w io.Writer) {
+	// Not yet clear what exactly might be most worth emitting here.
+}
+
+func (g mapGenerator) EmitNativeMaybe(w io.Writer) {
+	emitNativeMaybe(w, g.AdjCfg, g)
+}
+
+// --- type info --->
+
+func (g mapGenerator) EmitTypeConst(w io.Writer) {
+	doTemplate(`
+		// TODO EmitTypeConst
+	`, w, g.AdjCfg, g)
+}
+
+// --- TypedNode interface satisfaction --->
+
+func (g mapGenerator) EmitTypedNodeMethodType(w io.Writer) {
+	doTemplate(`
+		func ({{ .Type | TypeSymbol }}) Type() schema.Type {
+			return nil /*TODO:typelit*/
+		}
+	`, w, g.AdjCfg, g)
+}
+
+func (g mapGenerator) EmitTypedNodeMethodRepresentation(w io.Writer) {
+	// Perhaps surprisingly, the way to get the representation node pointer
+	//  does not actually depend on what the representation strategy is.
+	// REVIEW: this appears to be standard even across kinds; can we extract it?
+	doTemplate(`
+		func (n {{ .Type | TypeSymbol }}) Representation() ipld.Node {
+			return (*_{{ .Type | TypeSymbol }}__Repr)(n)
+		}
+	`, w, g.AdjCfg, g)
+}
+
+// --- Node interface satisfaction --->
+
+func (g mapGenerator) EmitNodeType(w io.Writer) {
+	// No additional types needed.  Methods all attach to the native type.
+}
+
+func (g mapGenerator) EmitNodeTypeAssertions(w io.Writer) {
+	doTemplate(`
+		var _ ipld.Node = ({{ .Type | TypeSymbol }})(&_{{ .Type | TypeSymbol }}{})
+		var _ schema.TypedNode = ({{ .Type | TypeSymbol }})(&_{{ .Type | TypeSymbol }}{})
+	`, w, g.AdjCfg, g)
+}
+
+func (g mapGenerator) EmitNodeMethodLookupString(w io.Writer) {
+	// What should be coercible in which directions (and how surprising that is) is an interesting question.
```

Comment:
Dear Reviewer, may I request your attention to this bit especially?

There's some _very_ interesting implications for PathSegments, how map keys work, and systemically emergent limitations that we need to think about.